sdk/java/src/main/java/com/gojek/feast/RequestUtil.java

```diff
@@ -17,47 +17,88 @@
+        featureRefStrings.stream()
+            .map(refStr -> parseFeatureRef(refStr, false))
+            .collect(Collectors.toList());
+    // apply project override if specified
+    if (!project.isEmpty()) {
+      featureRefs =
+          featureRefs.stream().map(ref -> ref.setProject(project)).collect(Collectors.toList());
+    }
 
-    for (String featureRefString : featureRefStrings) {
-      String project;
-      String name;
-      String[] projectSplit = featureRefString.split("/");
+    return featureRefs.stream().map(ref -> ref.build()).collect(Collectors.toList());
+  }
 
-      if (projectSplit.length == 1) {
-        project = defaultProject;
-        name = projectSplit[0];
-      } else if (projectSplit.length == 2) {
-        project = projectSplit[0];
-        name = projectSplit[1];
+  /**
+   * Parse a feature reference proto builder from the given featureRefString
+   *
+   * @param featureRefString string feature reference to parse from.
+   * @param ignoreProject whether to ignore project when parsing.
+   * @return a parsed {@link FeatureReference.Builder}
+   */
+  public static FeatureReference.Builder parseFeatureRef(
+      String featureRefString, boolean ignoreProject) {
+    featureRefString = featureRefString.trim();
+    if (featureRefString.isEmpty()) {
+      throw new IllegalArgumentException("Cannot parse a empty feature reference");
+    }
+    FeatureReference.Builder featureRef = FeatureReference.newBuilder();
+
+    // parse project if specified
+    if (featureRefString.contains("/")) {
+      if (ignoreProject) {
+        String[] projectSplit = featureRefString.split("/");
+        featureRefString = projectSplit[1];
       } else {
         throw new IllegalArgumentException(
-            String.format(
-                "Feature id '%s' has invalid format. Expected format: <project>/<feature-name>.",
-                featureRefString));
+            String.format("Unsupported feature reference: %s", featureRefString));
       }
+    }
 
-      if (project.isEmpty() || name.isEmpty() || name.contains(":")) {
-        throw new IllegalArgumentException(
-            String.format(
-                "Feature id '%s' has invalid format. Expected format: <project>/<feature-name>.",
-                featureRefString));
-      }
+    // parse featureset if specified
+    if (featureRefString.contains(":")) {
+      String[] featureSetSplit = featureRefString.split(":");
+      featureRef.setFeatureSetName(featureSetSplit[0]);
+      featureRefString = featureSetSplit[1];
+    }
+    featureRef.setName(featureRefString);
+    return featureRef;
+  }
 
-      featureRefs.add(FeatureReference.newBuilder().setName(name).setProject(project).build());
+  /**
+   * Render a feature reference as string.
+   *
+   * @param featureReference to render as string
+   * @return string represenation of feature reference.
+   */
+  public static String renderFeatureRef(FeatureReference featureReference) {
```

Comment:
Is `render` == `toString()`? I think if you want to introduce a new term then you should flesh out the comment a bit. It's a bit too vague.

Reply:
Corrected